elastic/kibana
# [Ingest Pipelines] Add geo-grid processor

Fixes: https://github.com/elastic/kibana/issues/154396

## Summary

Adds support for the [geo-grid processor](https://www.elastic.co/guide/en/elasticsearch/reference/master/ingest-geo-grid-processor.html).

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Ingest Pipelines` and create a new pipeline
3. Add new GeoGrid processor that has:
    * field: `geocell`
    * tile_type: `geo_hash`
    * target_format: `WKT`
4. Verify that:
    * The request is correctly generated
    * Testing the pipeline with the following document should produce a valid geo-grid output
 ```
[
  {
    "_index": "geocells",
    "_id": "1",
    "_version": 1,
    "_seq_no": 0,
    "_primary_term": 1,
    "found": true,
    "_source": {
      "geocell": "811fbffffffffff"
    }
  }
]
 ```
Should output:
```
{
  "docs": [
    {
      "doc": {
        "_index": "geocells",
        "_id": "1",
        "_version": "1",
        "_source": {
          "geocell": "BBOX (-177.50220526009798, -177.50220492482185, 6.123991962522268, 6.123991794884205)"
        },
        "_ingest": {
          "timestamp": "2023-04-18T12:52:15.397362Z"
        }
      }
    }
  ]
}
```

<details>
<summary>Screenshots</summary>

![Screenshot 2023-04-17 at 14 19 13](https://user-images.githubusercontent.com/1191206/232482060-479c670b-1130-4702-9ef1-1b815cd6174f.png)
![Screenshot 2023-04-17 at 14 18 49](https://user-images.githubusercontent.com/1191206/232482070-d00b3d66-1bc3-445f-ae57-30429cc7e7c5.png)
![Screenshot 2023-04-17 at 14 18 42](https://user-images.githubusercontent.com/1191206/232482075-2fb09d4f-694d-4630-b21d-49319897124f.png)


</details>
